src/components/currentLesson/CurrentLesson.tsx

```diff
@@ -0,0 +1,52 @@
+import './_currentLesson.scss';
+import { useEffect, useState } from "react"
+import { CurrentLessonProps } from "../../types/components/componentType"
+import defaultCoursePhoto from '../../img/defaultCoursePhoto.png';
+import axios from "axios";
+
+const CurrentLesson = () => {
+const [currentLessonProps, setCurrentLessonProps] = useState<CurrentLessonProps | null>(null);
+
+useEffect(() => {
+const fetchData = async () => {
+  try {
+    const response = await axios.get('URL/api/currentlesson');
+    const currentLessonData = response.data;
+
+    setCurrentLessonProps({
+      courseName: currentLessonData.courseName ?? "Science Basics",
+      lessonName: currentLessonData.lessonName ?? "Chemistry",
+      lessonCurrentNumber: currentLessonData.lessonCurrentNumber ?? 3,
+      allLessonsCount: currentLessonData.allLessonsCount ?? 6,
+    });
+  } catch (error) {
+    console.error('Error during fetching lesson details:', error);
+    setCurrentLessonProps({
+      courseName: "Name of the course",
+      lessonName: "Name of the lesson",
+      lessonCurrentNumber: 0,
+      allLessonsCount: 0,
+    });
+  }
+};
```

Comment:
```suggestion
const [currentLesson, setCurrentLesson] = useState<CurrentLesson>({
      courseName: "Name of the course",
      lessonName: "Name of the lesson",
      lessonCurrentNumber: 0,
      allLessonsCount: 0,
});

useEffect(() => {
const fetchData = async () => {
  try {
    const response = await axios.get('URL/api/currentlesson');
    const currentLessonData = response.data;

    setCurrentLessonProps({
      courseName: currentLessonData.courseName ?? "Science Basics",
      lessonName: currentLessonData.lessonName ?? "Chemistry",
      lessonCurrentNumber: currentLessonData.lessonCurrentNumber ?? 3,
      allLessonsCount: currentLessonData.allLessonsCount ?? 6,
    });
  } catch (error) {
    console.error('Error during fetching lesson details:', error);
  }
};
```
It's a better approach in React. If you don't get data from server, you will just have initial state in useState construction. For errors tou can create a separate state, in which you can set errors,  but here it's unnecessary.

Reply:
fixed

Reply:
Why you delete all logic with it in last commit?

Reply:
I have changed the way component gets all the necessary data. Instead of doing request by itself, component needs all the data at the initialization level.